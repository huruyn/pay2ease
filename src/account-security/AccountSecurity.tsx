/* eslint-disable react/prop-types */

import { Container, Row, Col } from 'react-bootstrap';
import CommonPageContainer from '../common/CommonPageContainer';

const CardIconPlaceholder = ({ text }: { text: string }) => (
  <Container
    style={{
      marginTop: '20px',
      border: '2px dashed black',
      borderRadius: '50%',
      minHeight: '80px',
      width: '100%',
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
    }}
  >
    <Row
      style={{
        fontSize: '12px',
        textAlign: 'center',
        width: '70px',
      }}
    >
      {text}
    </Row>
  </Container>
);

const IconButtonPlaceholder = ({ text }: { text: string }) => (
  <Row
    style={{
      fontSize: '12px',
      border: '2px dashed black',
      padding: '5px',
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      maxWidth: '200px',
    }}
  >
    {text}
  </Row>
);

const IconPlaceholder = ({ text }: { text: string }) => (
  <Row
    style={{
      fontSize: '12px',
      border: '2px dashed black',
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      maxWidth: '50px',
    }}
  >
    {text}
  </Row>
);

const SectionHeader: React.FC<{ title: string }> = ({ title }) => (
  <Row>
    <h4
      style={{
        marginTop: '50px',
        lineHeight: '40px',
        borderBottom: '3px solid black',
        padding: 0,
      }}
    >
      {title}
    </h4>
  </Row>
);

const CardDefaultProps = {
  cardTextSecondary: '',
  containerStyle: {},
};

/**
 * Represents a card component.
 * @param {string} cardName - Title.
 * @param {string} cardText - Body.
 * @param {string} cardTextSecondary - optional second <p> tag
 * @param {string} buttonIcon - *for now just string placeholder*
 * @param {string} cardIcon - *for now just string placeholder*
 * @param {Record<string, string | number>} containerStyle - Style Object}
 */
const Card: React.FC<{
  cardName: string;
  cardText: string;
  cardTextSecondary?: string;
  buttonIcon: string;
  cardIcon: string;
  containerStyle?: Record<string, string | number>;
}> = ({
  cardName,
  cardText,
  buttonIcon,
  cardIcon,
  cardTextSecondary,
  containerStyle,
}) => (
  <Col
    style={{
      marginTop: '14px',
      border: '2px solid rgba(0, 0, 0, 0.1)',
      minHeight: '220px',
      display: 'flex',
      justifyContent: 'space-between',
      flexDirection: 'column',
      ...containerStyle,
    }}
  >
    <Row>
      <div style={{ display: 'inline-block', width: '100px' }}>
        <CardIconPlaceholder text={cardIcon} />
      </div>
      <Col style={{ marginTop: '18px' }}>
        <h5 style={{ fontWeight: 600, paddingBottom: '15px' }}>{cardName}</h5>
        <p>{cardText}</p>
        <p>{cardTextSecondary}</p>
      </Col>
    </Row>
    <Col
      xs={12}
      style={{
        display: 'flex',
        justifyContent: 'flex-end',
        paddingRight: '25px',
        paddingBottom: '15px',
      }}
    >
      <IconButtonPlaceholder text={buttonIcon} />
    </Col>
  </Col>
);

Card.defaultProps = CardDefaultProps;

const AccountSecurity = () => (
  <div className="page-content">
    <CommonPageContainer title="Account Security">
      <Row>
        <Col>
          <Row style={{ alignItems: 'center' }}>
            <Col xs={8}>
              <h6
                style={{
                  fontWeight: 400,
                  marginBottom: 0,
                  padding: 0,
                  marginLeft: '-12px',
                }}
              >
                Review and update your security settings to protect your account.
              </h6>
            </Col>
            <Col style={{ display: 'flex', justifyContent: 'flex-end' }}>
              <IconButtonPlaceholder text="Online btn" />
            </Col>
          </Row>
        </Col>
        <SectionHeader title="Identity verification" />
      </Row>
      <Row>
        <Card
          cardName="Verification contact info"
          cardText="Keep your email and phone numbers up to date so we can properly verify your account."
          cardIcon="user icon"
          buttonIcon="Update btn"
          containerStyle={{ marginRight: '10px' }}
        />
        <Card
          cardName="Change password"
          cardText="Ensure you have a unique password to protect your account."
          buttonIcon="Update btn"
          cardIcon="pass icon"
          containerStyle={{ marginLeft: '10px' }}
        />
      </Row>
      <SectionHeader title="Get push verification codes to your device" />
      <Row>
        <Card
          cardName="Registered device: Samsung Duane's Note10+"
          cardText="This device is registered to receive your one-time verification codes by push notification. For your security, only one device can be registered at a time."
          cardTextSecondary="Note: Unregister your device if it gets lost or stolen or if it is no longer secure enough for identity verification."
          cardIcon="device icon"
          buttonIcon="Unregister device btn"
        />
      </Row>
      <SectionHeader title="Security Alerts" />
      <Row>
        <Card
          cardName="Fraud Prevention"
          cardText="Get alerts when there's unusual activity on your account by text, email, push notification or My Messages."
          buttonIcon="Update btn"
          cardIcon="security icon"
        />
      </Row>
      <Row
        style={{ marginTop: '53px', borderTop: '2px solid rgba(0, 0, 0, 0.1)' }}
      />
      <Row
        style={{
          marginTop: '8px',
          marginLeft: 0,
          display: 'flex',
          flexDirection: 'row',
          flexWrap: 'nowrap',
          alignItems: 'center',
        }}
      >
        <IconPlaceholder text="Icon" />
        <h5 style={{ paddingLeft: '20px', margin: 0, fontSize: '16px' }}>
          Note:
        </h5>
      </Row>
      <Row>
        <Col>
          <h6
            style={{
              marginLeft: '-12px',
              marginTop: '15px',
              fontSize: '14px',
              fontWeight: 400,
              color: 'rgba(0, 0, 0, 0.5)',
            }}
          >
            To learn more about ZippyX protects you from fraud, and how you can
            protect yourself, visit
            {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
            <a href="#"> Privacy and Security</a>
            .
          </h6>
        </Col>
      </Row>
    </CommonPageContainer>
  </div>
);

export default AccountSecurity;
