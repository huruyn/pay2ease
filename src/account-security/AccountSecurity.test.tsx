import AccountSecurity from './AccountSecurity';

const ReactTestRenderer = require('react-test-renderer');

describe('Legal Component', () => {
  it('matches the snapshot', () => {
    const tree = ReactTestRenderer.create(<AccountSecurity />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
