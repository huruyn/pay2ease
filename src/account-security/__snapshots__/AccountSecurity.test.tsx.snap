// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legal Component matches the snapshot 1`] = `
<div
  className="page-content"
>
  <div
    className="d-flex justify-content-between align-items-center"
  >
    <h3>
      Account Security
    </h3>
  </div>
  <div
    className="content-wrapper"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="row"
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <div
            className="col-8"
          >
            <h6
              style={
                Object {
                  "fontWeight": 400,
                  "marginBottom": 0,
                  "marginLeft": "-12px",
                  "padding": 0,
                }
              }
            >
              Review and update your security settings to protect your account.
            </h6>
          </div>
          <div
            className="col"
            style={
              Object {
                "display": "flex",
                "justifyContent": "flex-end",
              }
            }
          >
            <div
              className="row"
              style={
                Object {
                  "alignItems": "center",
                  "border": "2px dashed black",
                  "display": "flex",
                  "fontSize": "12px",
                  "justifyContent": "center",
                  "maxWidth": "200px",
                  "padding": "5px",
                }
              }
            >
              Online btn
            </div>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <h4
          style={
            Object {
              "borderBottom": "3px solid black",
              "lineHeight": "40px",
              "marginTop": "50px",
              "padding": 0,
            }
          }
        >
          Identity verification
        </h4>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
        style={
          Object {
            "border": "2px solid rgba(0, 0, 0, 0.1)",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginRight": "10px",
            "marginTop": "14px",
            "minHeight": "220px",
          }
        }
      >
        <div
          className="row"
        >
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100px",
              }
            }
          >
            <div
              className="container"
              style={
                Object {
                  "alignItems": "center",
                  "border": "2px dashed black",
                  "borderRadius": "50%",
                  "display": "flex",
                  "justifyContent": "center",
                  "marginTop": "20px",
                  "minHeight": "80px",
                  "width": "100%",
                }
              }
            >
              <div
                className="row"
                style={
                  Object {
                    "fontSize": "12px",
                    "textAlign": "center",
                    "width": "70px",
                  }
                }
              >
                user icon
              </div>
            </div>
          </div>
          <div
            className="col"
            style={
              Object {
                "marginTop": "18px",
              }
            }
          >
            <h5
              style={
                Object {
                  "fontWeight": 600,
                  "paddingBottom": "15px",
                }
              }
            >
              Verification contact info
            </h5>
            <p>
              Keep your email and phone numbers up to date so we can properly verify your account.
            </p>
            <p>
              
            </p>
          </div>
        </div>
        <div
          className="col-12"
          style={
            Object {
              "display": "flex",
              "justifyContent": "flex-end",
              "paddingBottom": "15px",
              "paddingRight": "25px",
            }
          }
        >
          <div
            className="row"
            style={
              Object {
                "alignItems": "center",
                "border": "2px dashed black",
                "display": "flex",
                "fontSize": "12px",
                "justifyContent": "center",
                "maxWidth": "200px",
                "padding": "5px",
              }
            }
          >
            Update btn
          </div>
        </div>
      </div>
      <div
        className="col"
        style={
          Object {
            "border": "2px solid rgba(0, 0, 0, 0.1)",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginLeft": "10px",
            "marginTop": "14px",
            "minHeight": "220px",
          }
        }
      >
        <div
          className="row"
        >
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100px",
              }
            }
          >
            <div
              className="container"
              style={
                Object {
                  "alignItems": "center",
                  "border": "2px dashed black",
                  "borderRadius": "50%",
                  "display": "flex",
                  "justifyContent": "center",
                  "marginTop": "20px",
                  "minHeight": "80px",
                  "width": "100%",
                }
              }
            >
              <div
                className="row"
                style={
                  Object {
                    "fontSize": "12px",
                    "textAlign": "center",
                    "width": "70px",
                  }
                }
              >
                pass icon
              </div>
            </div>
          </div>
          <div
            className="col"
            style={
              Object {
                "marginTop": "18px",
              }
            }
          >
            <h5
              style={
                Object {
                  "fontWeight": 600,
                  "paddingBottom": "15px",
                }
              }
            >
              Change password
            </h5>
            <p>
              Ensure you have a unique password to protect your account.
            </p>
            <p>
              
            </p>
          </div>
        </div>
        <div
          className="col-12"
          style={
            Object {
              "display": "flex",
              "justifyContent": "flex-end",
              "paddingBottom": "15px",
              "paddingRight": "25px",
            }
          }
        >
          <div
            className="row"
            style={
              Object {
                "alignItems": "center",
                "border": "2px dashed black",
                "display": "flex",
                "fontSize": "12px",
                "justifyContent": "center",
                "maxWidth": "200px",
                "padding": "5px",
              }
            }
          >
            Update btn
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <h4
        style={
          Object {
            "borderBottom": "3px solid black",
            "lineHeight": "40px",
            "marginTop": "50px",
            "padding": 0,
          }
        }
      >
        Get push verification codes to your device
      </h4>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
        style={
          Object {
            "border": "2px solid rgba(0, 0, 0, 0.1)",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginTop": "14px",
            "minHeight": "220px",
          }
        }
      >
        <div
          className="row"
        >
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100px",
              }
            }
          >
            <div
              className="container"
              style={
                Object {
                  "alignItems": "center",
                  "border": "2px dashed black",
                  "borderRadius": "50%",
                  "display": "flex",
                  "justifyContent": "center",
                  "marginTop": "20px",
                  "minHeight": "80px",
                  "width": "100%",
                }
              }
            >
              <div
                className="row"
                style={
                  Object {
                    "fontSize": "12px",
                    "textAlign": "center",
                    "width": "70px",
                  }
                }
              >
                device icon
              </div>
            </div>
          </div>
          <div
            className="col"
            style={
              Object {
                "marginTop": "18px",
              }
            }
          >
            <h5
              style={
                Object {
                  "fontWeight": 600,
                  "paddingBottom": "15px",
                }
              }
            >
              Registered device: Samsung Duane's Note10+
            </h5>
            <p>
              This device is registered to receive your one-time verification codes by push notification. For your security, only one device can be registered at a time.
            </p>
            <p>
              Note: Unregister your device if it gets lost or stolen or if it is no longer secure enough for identity verification.
            </p>
          </div>
        </div>
        <div
          className="col-12"
          style={
            Object {
              "display": "flex",
              "justifyContent": "flex-end",
              "paddingBottom": "15px",
              "paddingRight": "25px",
            }
          }
        >
          <div
            className="row"
            style={
              Object {
                "alignItems": "center",
                "border": "2px dashed black",
                "display": "flex",
                "fontSize": "12px",
                "justifyContent": "center",
                "maxWidth": "200px",
                "padding": "5px",
              }
            }
          >
            Unregister device btn
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <h4
        style={
          Object {
            "borderBottom": "3px solid black",
            "lineHeight": "40px",
            "marginTop": "50px",
            "padding": 0,
          }
        }
      >
        Security Alerts
      </h4>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
        style={
          Object {
            "border": "2px solid rgba(0, 0, 0, 0.1)",
            "display": "flex",
            "flexDirection": "column",
            "justifyContent": "space-between",
            "marginTop": "14px",
            "minHeight": "220px",
          }
        }
      >
        <div
          className="row"
        >
          <div
            style={
              Object {
                "display": "inline-block",
                "width": "100px",
              }
            }
          >
            <div
              className="container"
              style={
                Object {
                  "alignItems": "center",
                  "border": "2px dashed black",
                  "borderRadius": "50%",
                  "display": "flex",
                  "justifyContent": "center",
                  "marginTop": "20px",
                  "minHeight": "80px",
                  "width": "100%",
                }
              }
            >
              <div
                className="row"
                style={
                  Object {
                    "fontSize": "12px",
                    "textAlign": "center",
                    "width": "70px",
                  }
                }
              >
                security icon
              </div>
            </div>
          </div>
          <div
            className="col"
            style={
              Object {
                "marginTop": "18px",
              }
            }
          >
            <h5
              style={
                Object {
                  "fontWeight": 600,
                  "paddingBottom": "15px",
                }
              }
            >
              Fraud Prevention
            </h5>
            <p>
              Get alerts when there's unusual activity on your account by text, email, push notification or My Messages.
            </p>
            <p>
              
            </p>
          </div>
        </div>
        <div
          className="col-12"
          style={
            Object {
              "display": "flex",
              "justifyContent": "flex-end",
              "paddingBottom": "15px",
              "paddingRight": "25px",
            }
          }
        >
          <div
            className="row"
            style={
              Object {
                "alignItems": "center",
                "border": "2px dashed black",
                "display": "flex",
                "fontSize": "12px",
                "justifyContent": "center",
                "maxWidth": "200px",
                "padding": "5px",
              }
            }
          >
            Update btn
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
      style={
        Object {
          "borderTop": "2px solid rgba(0, 0, 0, 0.1)",
          "marginTop": "53px",
        }
      }
    />
    <div
      className="row"
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "flexWrap": "nowrap",
          "marginLeft": 0,
          "marginTop": "8px",
        }
      }
    >
      <div
        className="row"
        style={
          Object {
            "alignItems": "center",
            "border": "2px dashed black",
            "display": "flex",
            "fontSize": "12px",
            "justifyContent": "center",
            "maxWidth": "50px",
          }
        }
      >
        Icon
      </div>
      <h5
        style={
          Object {
            "fontSize": "16px",
            "margin": 0,
            "paddingLeft": "20px",
          }
        }
      >
        Note:
      </h5>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <h6
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.5)",
              "fontSize": "14px",
              "fontWeight": 400,
              "marginLeft": "-12px",
              "marginTop": "15px",
            }
          }
        >
          To learn more about ZippyX protects you from fraud, and how you can protect yourself, visit
          <a
            href="#"
          >
             Privacy and Security
          </a>
          .
        </h6>
      </div>
    </div>
  </div>
  <div
    className="footer-content-wrapper"
  >
    <div
      className="quick-links container"
    >
      <div
        className="row"
      >
        <h3
          className="col-sm-12"
        >
          Quick Links
        </h3>
      </div>
      <div
        className="row"
      >
        <div
          className="col-sm-2"
        >
          <ul>
            <li>
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Home
              </a>
            </li>
            <li>
              Please Review Our Legal Stuff by visiting our
               
              <a
                className="text-decoration-none"
                href="/legal"
                onClick={[Function]}
              >
                Privacy Policy | TOU | EULA | CardHolder T&C Pages
              </a>
            </li>
          </ul>
        </div>
        <div
          className="col-sm-2"
        >
          8465 W Sahara Ave. Las Vegas, NEVADA
          <ul>
            <li>
              <a
                href="tel:+17026231876"
              >
                1.702.623.1876
              </a>
            </li>
            <li>
              <a
                href="mailto:contact@zippy.cash"
              >
                contact@zippy.cash
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-12 col-sm-12 col-xs-12 text-left"
        >
          © 2021 Zippy Cash LLC  All rights reserved.
          <br />
          <br />
          The Program Manager for the Zippy Cash® Prepaid Visa Debit Card is PayQuicker, LLC
          <br />
          Zippy Cash® is a registered trademark of Zippy Cash, LLC
          <br />
          The Zippy Cash Prepaid Visa Debit Card is issued by Metropolitan Commercial Bank (Member FDIC) pursuant to a license from Visa U.S.A. Inc. “Metropolitan Commercial Bank” and “Metropolitan” are registered trademarks of Metropolitan Commercial Bank ©2014. BY USING THIS CARD YOU AGREE TO THE TERMS AND CONDITIONS OF THE CARDHOLDER AGREEMENT AND FEE SCHEDULE, IF ANY. All trademarks and service marks are the property of their respective owners.
          <br />
          In compliance with the USA PATRIOT Act* and other laws, anyone who is a US resident over 18 with a valid Social Security number and a verifiable physical US address (no PO boxes or APO addresses) can apply for an Zippy Cash Card. This information must be provided, along with other information pertinent to receiving approval, when applying for a Zippy Cash Card. Currently, we do not issue the Zippy Cash Card to residents of Puerto Rico, Guam, the Virgin Islands, American Samoa, or other US territories.
          <br />
          *Important information for opening a card account: To help the federal government fight the funding of terrorism and money laundering activities, the USA PATRIOT Act requires all financial institutions and their third parties to obtain, verify and record information that identifies each person who opens a card account. What this means for you: When you open a card account, we will ask for your name, address, date of birth, social security number or country identification number, and other information that will allows us to identify you. We may also request a copy of your driver’s license or other documentation bearing your photo as verification of your identity.
          <br />
          Apple and the Apple logo are trademarks of Apple Inc., registered in the U.S. and other countries. App Store is a service mark of Apple Inc. Google Play is a trademark of Google Inc. Data rates may apply. Samsung Pay is a trademark of Samsung Inc.
        </div>
      </div>
    </div>
    <footer
      className="container mb-3"
    >
      <div
        className="d-none d-md-block"
      >
        <div
          className="d-flex justify-content-between border-bottom"
        />
        <div
          className="d-flex justify-content-between border-top"
        >
          <div
            className=""
          >
            <div
              className="footer-link mt-1"
            >
              <a
                className="text-decoration-none"
                href="/personal-profile"
              >
                dev docs
              </a>
            </div>
            <div
              className="footer-link mt-1"
            >
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Lore
              </a>
            </div>
            <div
              className="footer-link mt-1"
            >
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Ipsum
              </a>
            </div>
            <div
              className="footer-link mt-1"
            >
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Nullam Atoidio
              </a>
            </div>
          </div>
          <div
            className="copyright mt-1"
          >
            <span>
              © 2021 Lorem ipsum dolor sit amet, consectetuer
            </span>
          </div>
        </div>
      </div>
      <div
        className="d-block d-md-none"
      >
        <div
          className="d-flex border-bottom"
        >
          <div
            className="d-flex flex-column me-2"
          >
            <div
              className="footer-link mt-1"
            >
              <a
                className="text-decoration-none"
                href="/personal-profile"
              >
                dev docs
              </a>
            </div>
            <div
              className="footer-link mb-1"
            >
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Lore
              </a>
            </div>
            <div
              className="footer-link mb-1"
            >
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Ipsum
              </a>
            </div>
            <div
              className="footer-link mb-1"
            >
              <a
                className="text-decoration-none"
                href="/"
                onClick={[Function]}
              >
                Nullam Atoidio
              </a>
            </div>
          </div>
        </div>
        <div
          className="d-flex justify-content-end border-top"
        >
          <div
            className="copyright mt-2 text-end small"
          >
            <span>
              © 2021 Lorem ipsum dolor sit amet, consectetuer
            </span>
          </div>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
